example/src/SdkMethodSections/DeviceIDSection.tsx

```diff
@@ -1,20 +1,32 @@
 import * as React from 'react';
-import { Button } from 'react-native-elements';
-import { useAmplitude } from '../utils/amplitude';
-
+import { Button, Input } from 'react-native-elements';
 import { SdkSectionLayout } from './SdkSectionLayout';
+import { useAmplitude } from '../utils/amplitude';
 
 export const DeviceIDSection = () => {
   const amplitude = useAmplitude();
+  const [currentDeviceId, setCurrentDeviceId] = React.useState('');
+  const [loading, updateLoading] = React.useState(false);
+
+  const handleOnPress = () => {
+    amplitude.regenerateDeviceId();
+    updateLoading(true);
+  };
+
+  const getDeviceId = React.useCallback(async () => {
+    const deviceId = await amplitude.getDeviceId();
+    setCurrentDeviceId(deviceId);
+    updateLoading(false);
+  }, [amplitude]);
+
+  React.useEffect(() => {
+    getDeviceId();
+  }, [getDeviceId, loading]);
 
   return (
     <SdkSectionLayout heading={'Device ID'}>
-      <Button
-        title={'Regenerate Device ID'}
-        onPress={() => {
-          amplitude.regenerateDeviceId();
-        }}
-      />
+      <Input value={currentDeviceId} editable={false} />
```

Comment:
Might be good to wrap in a view without pointer events, ie https://stackoverflow.com/a/54127319